src/scripts/markdown-checker.js

```diff
@@ -0,0 +1,219 @@
+const fs = require("fs")
+const path = require("path")
+const matter = require("gray-matter")
+const argv = require("minimist")(process.argv.slice(2))
+
+const LANG_ARG = argv.lang || null
+const PATH_TO_INTL_MARKDOWN = "./src/content/translations/"
+const PATH_TO_ALL_CONTENT = "./src/content/"
+const TUTORIAL_DATE_REGEX = new RegExp("\\d{4}-\\d{2}-\\d{2}")
+const WHITE_SPACE_IN_LINK_TEXT = new RegExp(
+  "\\[\\s.+\\]\\( | \\[.+\\s\\]\\(",
+  "g"
+)
+const BROKEN_LINK_REGEX = new RegExp(
+  "\\[[^\\]]+\\]\\([^\\)\\s]+\\s[^\\)]+\\)",
+  "g"
+)
+const HTML_TAGS = ["</code", "</p>"]
+const SPELLING_MISTAKES = [
+  "Ethreum",
+  "Etherum",
+  "Etherium",
+  "Etheruem",
+  "Etereum",
+  "Eterium",
+  "Etherem",
+  "Etheerum",
+  "Ehtereum",
+  "Eferum",
+]
+const CASE_SENSITVE_SPELLING_MISTAKES = ["Thereum", "Metamask", "Github"]
+// Ideas:
+// Regex for explicit lang path (e.g. /en/) && for glossary links (trailing slash breaks links e.g. /glossary/#pos/ doesn't work)
+// We should have case sensitive spelling mistakes && check they are not in links.
+
+const langsArray = fs.readdirSync(PATH_TO_INTL_MARKDOWN)
+langsArray.push("en")
+
+function getAllMarkdownPaths(dirPath, arrayOfMarkdownPaths = []) {
+  let files = fs.readdirSync(dirPath)
+
+  arrayOfMarkdownPaths = arrayOfMarkdownPaths || []
+
+  for (const file of files) {
+    if (fs.statSync(dirPath + "/" + file).isDirectory()) {
+      arrayOfMarkdownPaths = getAllMarkdownPaths(
+        dirPath + "/" + file,
+        arrayOfMarkdownPaths
+      )
+    } else {
+      const filePath = path.join(dirPath, "/", file)
+
+      if (filePath.includes(".md")) {
+        arrayOfMarkdownPaths.push(filePath)
+      }
+    }
+  }
+
+  return arrayOfMarkdownPaths
+}
+
+function sortMarkdownPathsIntoLanguages(files) {
+  const languages = langsArray.reduce((accumulator, value) => {
+    return { ...accumulator, [value]: [] }
+  }, {})
+
+  for (const file of files) {
+    const isTranslation = file.includes("/translations/")
+    const langIndex = file.indexOf("/translations/") + 14
+    const isFourCharLang = file.includes("pt-br") || file.includes("zh-tw")
+    const charactersToSlice = isFourCharLang ? 5 : 2
+
+    const lang = isTranslation
+      ? file.slice(langIndex, langIndex + charactersToSlice)
+      : "en"
+
+    if (LANG_ARG) {
+      if (LANG_ARG === lang) {
+        languages[lang].push(file)
+      }
+    } else {
+      languages[lang].push(file)
+    }
+  }
+
+  return languages
+}
+
+function processFrontmatter(path, lang) {
```

Comment:
@minimalsm 

We can directly call this function with the path of the staged markdown to process its frontmatter.

Reply:
We can either use:
- `git diff`
- [A Node Wrapper over `git diff`](https://www.npmjs.com/package/staged-git-files)
- [nodegit](https://github.com/nodegit/nodegit)

to get the staged files along with their paths.